### PR TITLE
Chore: Add dashboard UID as query parameter of Get annotation endpoint

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #52459

**Special notes for your reviewer**:

### CODE REVIEW

**Review by chri2547 — COMMENTED** [review]
The docs look fine. Please go ahead and merge pending code review and approval.

**Review by zserge — APPROVED** [review]

File: pkg/api/annotations.go [hunk 1]
```diff
@@ -34,6 +35,16 @@ func (hs *HTTPServer) GetAnnotations(c *models.ReqContext) response.Response {
 		SignedInUser: c.SignedInUser,
 	}
 
+	// When dashboard UID present in the request, we ignore dashboard ID
```
Comment (papagian) [inline pkg/api/annotations.go:38]: should we ignore the `dashboardId` or return for both?
if we decide the former then we need to mention it in the docs/spec

File: pkg/api/annotations.go [hunk 2]
```diff
@@ -34,6 +35,16 @@ func (hs *HTTPServer) GetAnnotations(c *models.ReqContext) response.Response {
...
+	if query.DashboardUid != "" {
+		dq := models.GetDashboardQuery{Uid: query.DashboardUid, OrgId: c.OrgId}
+		err := hs.DashboardService.GetDashboard(c.Req.Context(), &dq)
+		if err != nil {
+			return response.Error(http.StatusBadRequest, "Invalide dashboard UID in the request", err)
```
Comment (papagian) [inline pkg/api/annotations.go:43]:
```suggestion
			return response.Error(http.StatusBadRequest, "Invalid dashboard UID in the request", err)
```

**Review by papagian — APPROVED** [review]

File: pkg/api/annotations.go (hunk 1 above)
Comment (ying-jeanne) [reply pkg/api/annotations.go:38]: Let me update the doc

**Pull request merged**